### PR TITLE
TYP: mostly missing return annotations in IO

### PR DESCRIPTION
**Pull request by twoertwein**

### CODE REVIEW

File: pandas/io/clipboards.py [hunk 1]
```diff
@@ -83,7 +85,9 @@ def read_clipboard(sep=r"\s+", **kwargs):  # pragma: no cover
     return read_csv(StringIO(text), sep=sep, **kwargs)
 
 
-def to_clipboard(obj, excel=True, sep=None, **kwargs):  # pragma: no cover
+def to_clipboard(
+    obj, excel: bool | None = True, sep: str | None = None, **kwargs
+) -> None:  # pragma: no cover
```
Comment (twoertwein) [inline pandas/io/clipboards.py:90]: is `# pragma: no cover` for codecov? Does it need to be on the `def to_clipboard`-line?

File: pandas/io/common.py [hunk 2]
```diff
@@ -170,7 +170,7 @@ def _expand_user(filepath_or_buffer: str | BaseBufferT) -> str | BaseBufferT:
     return filepath_or_buffer
 
 
-def validate_header_arg(header) -> None:
+def validate_header_arg(header: object) -> None:
```
Comment (jbrockmendel) [inline pandas/io/common.py:173]: should be `Any`?

File: pandas/io/html.py [hunk 3]
```diff
@@ -116,6 +116,7 @@ def _get_skiprows(skiprows: int | Sequence[int] | slice | None):
         start, step = skiprows.start or 0, skiprows.step or 1
         return list(range(start, skiprows.stop, step))
     elif isinstance(skiprows, numbers.Integral) or is_list_like(skiprows):
+        assert skiprows is not None
```
Comment (jbrockmendel) [inline pandas/io/html.py:119]: can you make this a `cast`?  that way mypy will alert us once it is unnecessary

File: pandas/io/common.py (hunk 2 above)
Comment (twoertwein) [reply pandas/io/common.py:173]: Typeshed recommends to use `Any` when it is too difficult to create a type annotations. `object` should be used if literally any object can be accepted.

https://github.com/python/typeshed/blob/master/CONTRIBUTING.md#conventions

> Note that Any is not the correct type to use if you want to indicate that some function can accept literally anything: in those cases use object instead.

**Pull request merged**